### PR TITLE
Add support of new GetSystemTime RPC

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Added implementation of onSystemTimeReady().
Added onSystemTimeReady notification right after onReady().
Added implemetation of GetSystemTime() to response on approriate SDL request.

Related tasks : [APPLINK-29396](https://adc.luxoft.com/jira/browse/APPLINK-29396), [APPLINK-29386](https://adc.luxoft.com/jira/browse/APPLINK-29386)

### CODE REVIEW

**Review by OHerasym — APPROVED** [review]

File: ffw/BasicCommunicationRPC.js [hunk 1]
```diff
@@ -901,6 +954,18 @@ FFW.BasicCommunication = FFW.RPCObserver
         this.client.send(JSONMessage);
       },
       /**
+       * notify SDL about readiness to provide system time.
+       * BasicCommunication should be sunscribed to this notification
```
Comment (pvvasilev) [inline ffw/BasicCommunicationRPC.js]: @AKalinich-Luxoft typo "subscribed"

**Review by pvvasilev — APPROVED** [review]

**Review by wolfylambova22 — APPROVED** [review]

File: ffw/BasicCommunicationRPC.js [hunk 2]
```diff
@@ -728,6 +731,55 @@ FFW.BasicCommunication = FFW.RPCObserver
...
+        var current_tz_hour = Math.floor(-current_time.getTimezoneOffset() / 60);
+        var current_tz_minute =
+            (-current_time.getTimezoneOffset()) - (current_tz_hour * 60);
+        if (current_tz_minute < 0) {
+          current_tz_minute += 60;
+          if (current_tz_hour == 14)
+            current_tz_hour = -12;
+          else
+            current_tz_hour++;
+        }
```
Comment (Kozoriz) [inline ffw/BasicCommunicationRPC.js]: @AKalinich-Luxoft , can we use : 

> var tzo = -current_time.getTimezoneOffset();
> var tz_hour = tzo / 60;
> var tz_min = tzo % 60; 

?

**Review by Kozoriz — CHANGES_REQUESTED** [review]

**Review by AByzhynar — APPROVED** [review]

**Comment by AByzhynar** [comment]
@AKalinich-Luxoft Sorry I don't know JS

File: ffw/BasicCommunicationRPC.js (hunk 1 above)
Comment (AKalinich-Luxoft) [reply ffw/BasicCommunicationRPC.js]: @pvvasilev, thanks, fixed in [902bd8d](https://github.com/CustomSDL/sdl_hmi/pull/4/commits/902bd8d5e6a5cc201de393e3f41741f350a08d10)

File: ffw/BasicCommunicationRPC.js (hunk 2 above)
Comment (AKalinich-Luxoft) [reply ffw/BasicCommunicationRPC.js]: @Kozoriz, yes, it's good idea. But there were some clarifications for this function and there is no need to calculate this fields anymore (see [APPLINK-33109](https://adc.luxoft.com/jira/browse/APPLINK-33109)). I've changed it in [902bd8d](https://github.com/CustomSDL/sdl_hmi/pull/4/commits/902bd8d5e6a5cc201de393e3f41741f350a08d10)

**Review by anosach-luxoft — APPROVED** [review]

**Review by Kozoriz — APPROVED** [review]

**Pull request merged**